app/src/main/java/org/schabi/newpipe/util/StreamDialogEntry.java

```diff
@@ -1,21 +1,61 @@
 package org.schabi.newpipe.util;
 
 import android.content.Context;
+import android.widget.Toast;
 
 import androidx.fragment.app.Fragment;
 
 import org.schabi.newpipe.R;
 import org.schabi.newpipe.extractor.stream.StreamInfoItem;
 import org.schabi.newpipe.local.dialog.PlaylistAppendDialog;
+import org.schabi.newpipe.player.MainPlayer;
+import org.schabi.newpipe.player.helper.PlayerHolder;
 import org.schabi.newpipe.player.playqueue.SinglePlayQueue;
 
+import java.util.ArrayList;
 import java.util.Collections;
 
 public enum StreamDialogEntry {
     //////////////////////////////////////
     // enum values with DEFAULT actions //
     //////////////////////////////////////
 
+    /**
+     * Enqueues the stream automatically to the current PlayerType.<br>
+     * <br>
+     * Info: Add this entry within showStreamDialog.
+     */
+    enqueue_stream(R.string.enqueue_stream, (fragment, item) -> {
+        final MainPlayer.PlayerType type = PlayerHolder.getType();
+
+        if (type == null) {
+            // This code shouldn't be reached since the checks for appending this entry should be
+            // done within the showStreamDialog calls.
+            Toast.makeText(fragment.getContext(),
+                    "No player currently playing", Toast.LENGTH_SHORT).show();
+            return;
+        }
+        switch (type) {
+            case AUDIO:
+                NavigationHelper.enqueueOnBackgroundPlayer(fragment.getContext(),
+                        new SinglePlayQueue(item), false);
+                break;
+            case POPUP:
+                NavigationHelper.enqueueOnPopupPlayer(fragment.getContext(),
+                        new SinglePlayQueue(item), false);
+                break;
+            case VIDEO:
+                NavigationHelper.enqueueOnVideoPlayer(fragment.getContext(),
+                        new SinglePlayQueue(item), false);
+                break;
+            default:
+                // Same as above, but keep it for now for debugging.
+                Toast.makeText(fragment.getContext(),
+                        "Unreachable code executed", Toast.LENGTH_SHORT).show();
+                break;
+        }
```

Comment:
I think you can replace this part with something like this, and call it a day. As you said the `type` can never theoretically be `null` or an invalid value, so we can just include that impossible situation into the condition for opening the main video player (enqueueing when there is no player does not raise errors, but just opens that player).
```suggestion
        if (type == AUDIO) {
            NavigationHelper.enqueueOnBackgroundPlayer(fragment.getContext(),
                    new SinglePlayQueue(item), false);
        } else if (type == POPUP) {
            NavigationHelper.enqueueOnPopupPlayer(fragment.getContext(),
                    new SinglePlayQueue(item), false);
        } else /* type == VIDEO */ {
                NavigationHelper.enqueueOnVideoPlayer(fragment.getContext(),
                        new SinglePlayQueue(item), false);
        }
```

Reply:
I kept it before reviews, when there might occur something unexpected, but yeah, I'll make it more compact.